libavfilter/dnn_filter_common.c

```diff
@@ -17,6 +17,36 @@
  */
 
 #include "dnn_filter_common.h"
+#include "libavutil/avstring.h"
+
+static char **separate_expression(DnnContext *ctx, const char *expr, const char *val_sep, int *separated_nb)
+{
+    char *val, **parsed_vals = NULL;
+    int val_num = 0;
+    if (!expr || !val_sep || !separated_nb) {
+        return NULL;
+    }
+
+    if (!parsed_vals)
+        parsed_vals = av_mallocz(sizeof(*parsed_vals));
+
+    do {
+        val = av_get_token(&expr, val_sep);
+        if(strlen(val)) {
```

Comment:
if (val)  ?

Reply:
since the av_get_token will return a string with no letter in it, like "", if the expr begins with the val_sep &, like "&input_1&input_2". it is uncommon, but maybe using the strlen() here is better?

Reply:
maybe we don't need to handle the options smartly, for such case, we see three strings: "", "input_1" and "input_2"

Reply:
tf backend will report error (Could not find output "" in model ), since it think the "" is a name of output node. obviously, a string without any letter cannot be a output name.

Reply:
that's user's responsibility to fix it, we'd report it, instead of fixing it internally.

Reply:
oh, that make sense, i will modify it